config/overlays/kind/generate-tls.sh

```
@@ -0,0 +1,50 @@
+# Directory where certificates will be stored (relative to script location)
+SCRIPT_DIR=$(dirname "$(realpath "$0")")
+CERT_FILE="$SCRIPT_DIR/tls.crt"
+KEY_FILE="$SCRIPT_DIR/tls.key"
+
+# Generate certificate and key if they don't exist or if explicitly requested
+if [ "${1:-}" = "generate" ] || [ ! -f "$CERT_FILE" ] || [ ! -f "$KEY_FILE" ]; then
+    echo "Generating new TLS certificate and key..." >&2
+
+    # Create temporary directory for generation
+    TEMP_DIR=$(mktemp -d)
+    trap "rm -rf $TEMP_DIR" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)